bait-files/src/test/java/com/google/sps/servlets/RequestsServletTest.java

```diff
@@ -0,0 +1,128 @@
+    objectify = ObjectifyService.begin();
+  }
+
+  @After
+  public void tearDown() throws InterruptedException, IOException, TimeoutException {
+    objectify.close();
+  }
+
+  @AfterClass
+  public static void oneTimeTearDown() throws InterruptedException, IOException, TimeoutException {
+    datastoreHelper.stop();
+  }
+
+  @Test
+  public void doGet_whenNoRequestInDb_returnsNothing() throws IOException, ServletException {
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    HttpServletResponse response = mock(HttpServletResponse.class);
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(writer);
+
+    new RequestsServlet().doGet(request, response);
+
+    String rawJsonResponse = stringWriter.toString();
+    assertTrue(rawJsonResponse.startsWith("[]"));
+  }
+
+  @Test
+  public void doGet_whenOneRequestInDb_returnsOneRequest() throws IOException, ServletException {
+    ObjectifyService.ofy()
+        .save()
+        .entity(new Request(14L, "4", "human", "human47", "human47@gmail.com", "2930 pearl street",
+            "no_image", "719-325-6872", "some notes"))
+        .now();
+
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    HttpServletResponse response = mock(HttpServletResponse.class);
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(writer);
+
+    new RequestsServlet().doGet(request, response);
+
+    String rawJsonResponse = stringWriter.toString();
+    assertTrue(rawJsonResponse.startsWith("[{\"requestId\":14"));
+  }
+
+  @Test
+  public void doGet_returnsOnlyPendingRequests() throws IOException, ServletException {
+    ObjectifyService.ofy()
+        .save()
+        .entity(new Request(14L, "4", "human", "human47", "human47@gmail.com", "2930 pearl street",
+            "no_image", "719-325-6872", "some notes"))
+        .now();
+
+    ObjectifyService.ofy()
+        .save()
+        .entity(new Request(19L, "4", "human", "human47", "human47@gmail.com", "2930 pearl street",
+            "no_image", "719-325-6872", "some notes"))
+        .now();
+
+    Request pendingRequest =
+        ObjectifyService.ofy().load().type(Request.class).id(Long.parseLong("19")).now();
+    pendingRequest.status = Status.COMPLETED;
+
+    ObjectifyService.ofy().save().entity(pendingRequest).now();
+
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    HttpServletResponse response = mock(HttpServletResponse.class);
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(writer);
+
+    new RequestsServlet().doGet(request, response);
+
+    String rawJsonResponse = stringWriter.toString();
+    assertTrue(rawJsonResponse.startsWith("[{\"requestId\":14"));
```

Comment:
Same comment as elsewhere... Does this actually confirm that the other request was not chosen?

Reply:
Changed it hopefully it's more clear now.